frontends/web/src/api/coins.ts

```diff
@@ -46,3 +46,16 @@ export type TAmount = {
 export const parseExternalBtcAmount = (amount: string): Promise<TAmount> => {
   return apiGet(`coins/btc/parse-external-amount?amount=${amount}`);
 };
+
+export type AddressSignResponse = {
+  success: boolean;
+  signature: string;
+  address: string;
+  errorMessage?: string;
+  errorCode?: BackendErrorCode;
+}
+
```

Comment:
Please change to

```ts
export type AddressSignResponse = {
  success: true;
  signature: string;
  address: string;
} | {
  success: false;
  errorMessage?: string;
  errorCode: BackendErrorCode;
}
```

this is just an example, please adjust the optional fields as the backend returns those. Reason this way we can tell TypeScript that there is always a errorCode if success is false. Or that there is a signature if success true. etc.

Reply:
Done, thanks!